content/blog/2020/07/2020-07-30-github-checks-api-plugin-coding-phase-2.adoc

```diff
@@ -0,0 +1,149 @@
+            cpd(pattern: 'target/cpd.xml')],
+            qualityGates: [[threshold: 1, type: 'TOTAL', unstable: true]]
+    }
+}
+----
+
+_For more about the pipeline usage of warnings-ng plugin, please see link:https://github.com/jenkinsci/warnings-ng-plugin/blob/master/doc/Documentation.md#pipeline-configuration[the official documentation]._
+
+However, if you don't want to publish the warnings to GitHub, you can definitely disable it by using a boolean field `skipPublishingChecks`, e.g.
+
+[source, groovy]
+----
+recordIssues enabledForFailure: true, tools: [java(), javaDoc()], skipPublishingChecks: true
+----
+
+== Coverage Checks
+
+The coverage checks are achieved by consuming the API in link:https://plugins.jenkins.io/code-coverage-api/[Code Coverage API plugin].
+First, in the conversation tab of a PR, users will be able to see the summary about the coverage and the comparing with previous builds.
+
+image::/images/post-images/2020/07-github-checks-api-plugin-coding-phase-2/coverage-summary.png[Coverage Summary]
+
+Besides, the `Details` page will contain some other things:
+
+* The links to the reference build, including the target branch build from the master branch and the last successful build from this branch
+* Coverage healthy score (the default value is 100% if the threshold is not configured)
+* Coverages and trends of different types in table format
+
+image::/images/post-images/2020/07-github-checks-api-plugin-coding-phase-2/coverage-details.png[Coverage Details]
+
+The link:https://github.com/jenkinsci/code-coverage-api-plugin/pull/169[pull request] for this feature will soon be merged and will be included in the next release of Coverage Checks API plugin. After that, you can use it by adding the below section to your pipeline script:
+
+[source, groovy]
+----
+node {
+    stage ('Checkout') {
+        checkout scm
+    }
+
+    stage ('Line and Branch Coverage') {
+        withMaven {
+            sh "mvn -V -U -e jacoco:prepare-agent test jacoco:report -Dmaven.test.failure.ignore"
+        }
+        publishCoverage adapters: [jacocoAdapter('**/*/jacoco.xml')], sourceFileResolver: sourceFiles('STORE_ALL_BUILD')
+    }
+}
+----
+
+Like the warning checks, you can also disable the coverage checks by setting the field `skipPublishingChecks`, e.g.
+
+[source, groovy]
+----
+publishCoverage adapters: [jacocoAdapter('**/*/jacoco.xml')], sourceFileResolver: sourceFiles('STORE_ALL_BUILD'), skipPublishingChecks: true
+----
+
+== Next Phase
+
+In the next phase, we will turn our attention back to link:https://plugins.jenkins.io/checks-api/[Checks API Plugin] and link:https://plugins.jenkins.io/github-checks/[GitHub Checks Plugin]:
+
+* Pipeline support will be added, so users can publish checks directly in a pipeline script without requiring a consumer plugin that supports the checks.
+* Re-run request for Jenkins build through Checks API will be added as well. 
+
+Hopefully, we will also release the 1.0.0 version for both plugins.
```

Comment:
Reference phase 3? I'd guess hope only applies to the timeline, not to a fact of the release